src/tasks/trainerUtils.py

```diff
@@ -227,180 +156,287 @@ def parse_ev_script_file(file_path):
+        return trainer
+
+def get_zoneID(areaName):
+    for places in areas:
+        if areaName in places:
+            zoneID = int(areas.index(places) - 1)
+            return zoneID
+        else:
+            continue
+
+def get_multi_trainer_data(file_path, areaName, zoneID, trainerID1, trainerID2, trainerID3):
+    trainers = []
+    if trainerID3.isnumeric():
+        
+        trainers += get_support_trainers_data(file_path, areaName, "male", zoneID)
+        trainers += get_support_trainers_data(file_path, areaName, "female", zoneID)
+        trainer2, trainer3 = get_multi_trainers(trainerID2, trainerID3, zoneID, "Multi")
+        trainers.append(trainer2)
+        trainers.append(trainer3)
+    elif trainerID3[0] == "@":
+
+        trainers += get_support_trainers_data(file_path, areaName, "male", zoneID)
+        trainers += get_support_trainers_data(file_path, areaName, "female", zoneID)
+        team_galactic_lookup = f"pos_{areaName.lower()}_gingakanbu"
+
+        temp_enemy_IDs = parse_random_teams(file_path, team_galactic_lookup, 2, "Evil")
+        if temp_enemy_IDs:
+            trainerID2 = temp_enemy_IDs[0]
+            trainerID3 = temp_enemy_IDs[1]
+            trainer2, trainer3 = get_multi_trainers(trainerID2, trainerID3, zoneID, "Multi")
+            trainers.append(trainer2)
+            trainers.append(trainer3)
+        else:
+            print("Unable to find Multi Trainer (variable) opponent teams", areaName, substring)
+
+    return trainers
+
+def get_random_team_data(file_path, areaName, zoneID, trainerID1, trainerID2, lookup, team_num):
+    trainers = []
+    if "Masters" not in lookup and "Celebi" not in lookup:
+        temp_trainer_IDs = parse_random_teams(file_path, lookup, team_num, None)
+        type = lookup.split("_")[-1].strip('"')
+        if "barry" in lookup:
+            for ID in temp_trainer_IDs:
+                trainer = get_single_trainer(zoneID, ID, temp_trainer_IDs, type)
+                trainers.append(trainer)
+            return trainers
+        for ID in temp_trainer_IDs:
+            trainer = get_single_trainer(zoneID, ID, temp_trainer_IDs, None)
+            trainers.append(trainer)
+        return trainers
+    if "Masters" in lookup:
+        temp_trainer_IDs = parse_random_teams(file_path, "", len(team_num), lookup)
+        for ID in temp_trainer_IDs:
+            master_trainer_name = team_num[temp_trainer_IDs.index(ID)]
+            trainer = get_single_trainer(zoneID, ID, temp_trainer_IDs, team_num[temp_trainer_IDs.index(ID)])
+            trainers.append(trainer)
+        return trainers
+    if "Celebi" in lookup:
+        temp_trainer_IDs = parse_random_teams(file_path, "", team_num, lookup)
+        for ID in temp_trainer_IDs:
+            trainer = get_single_trainer(zoneID, ID, temp_trainer_IDs, lookup)
+            trainers.append(trainer)
+        return trainers
+
+def get_assorted_trainer_data(file_path, areaName, zoneID, trainerID1, trainerID2, args):
```

Comment:
Mostly the same issues as above. Focus on modularity and error handling with this one.

Rough example
```py
def get_assorted_trainer_data(file_path, area_name, zone_id, trainer_id1, trainer_id2, args):
    trainers = []
    cyrus_lookup = f"ev_{area_name.lower()}_randomteam_cyrus"
    master_trainer_types = ["fire", "water", "electric", "grass", "ice", "fighting", "poison", "ground", "flying", "psychic", "bug", "rock", "ghost", "dragon", "dark", "steel", "fairy"]
    starters = ["piplup", "turtwig", "chimchar"]

    for starter in starters:
        rival_lookup = f"ev_{area_name.lower()}_randomteam_barry_{starter}"
        rival_teams = get_random_team_data(file_path, area_name, zone_id, trainer_id1, trainer_id2, rival_lookup, 4)
        add_trainers(trainers, rival_teams)

    cyrus_teams = get_random_team_data(file_path, area_name, zone_id, trainer_id1, trainer_id2, cyrus_lookup, 4)
    add_trainers(trainers, cyrus_teams, "Cyrus")

    master_teams = get_master_trainer_data(file_path, area_name, zone_id, trainer_id1, trainer_id2, master_trainer_types)
    add_trainers(trainers, master_teams, "Master Trainer")

    celebi_teams = get_random_team_data(file_path, area_name, zone_id, trainer_id1, trainer_id2, "Celebi", 7)
    add_trainers(trainers, celebi_teams, "Celebi")

    if not trainers:
        print("Lucas and Dawn's Single Battles are not yet supported:", area_name, args[0])
         //Removed comment in example
    return trainers


def add_trainers(trainers, new_trainers, trainer_type=None):
    for trainer in new_trainers:
        if trainer_type:
            trainer["trainer_type"] = trainer_type
        trainers.append(trainer)


def get_master_trainer_data(file_path, area_name, zone_id, trainer_id1, trainer_id2, trainer_types):
    trainers = []
    for trainer_type in trainer_types:
        lookup = f"ev_{area_name.lower()}_randomteam_barry_{trainer_type}"
        trainer_teams = get_random_team_data(file_path, area_name, zone_id, trainer_id1, trainer_id2, lookup, 4)
        add_trainers(trainers, trainer_teams, f"Master Trainer - {trainer_type.capitalize()}")
    return trainers
```